apps/src/lab2/types.ts

```diff
@@ -181,6 +181,8 @@ export interface LevelProperties {
   validationFile?: ProjectFile;
   useSecondaryFinishButton?: boolean;
   enableMicroBit?: boolean;
+  serializedMaze?: MazeCell[][];
```

Comment:
> Do you see us eventually migrating javalab to lab2?
Hopefully yes!

In terms of lab-specific properties, I'd rather have them here and optional rather than have to keep casting the level properties to different sub-types. There could be a refactor down the line if things get unwieldy where we do some fancy stuff (maybe with TypeScript generics?) but for now I think it's ok to have this mega-type. I can add comments to the file about which are python lab specific. There's also a fair amount of music lab specific types in here too.

Reply:
> I can add comments to the file about which are python lab specific. There's also a fair amount of music lab specific types in here too.

Makes sense! And I think a comment here would be helpful. Didn't realize that there were musiclab-specific types here as well. Maybe a comment can be added eventually for these? cc: @sanchitmalhotra126

Reply:
Re: types, yeah that's fair - we do have those lab-specific subtypes and we keep casting when referencing in components which is a bit cumbersome; I do like the option to extend the LevelProperties interface with non-optional lab-specific keys (e.g. in Panels) but I agree that doing some fancy with TypeScript generics would be ideal.

If anything, I feel like we kind of have a separation where very lab-specific fields go in the subtype, vs "_technically_ shareable but only used by one lab in practice" fields end up in level properties? All this to say, Alice yeah I agree maybe we just need comments for any specific types here.

Reply:
I was wrong, we do already have a subtype for video levels that we are casting to, but then we have some other things (like `useSecondaryFinishButton`) that are only used in Music Lab (and Music Lab doesn't have a subtype). So we should probably unify on a strategy here long-term. I think I'll leave that as a follow-up though.

Reply:
Yeah, we do have a `MusicLevelData` that extends `levelProperties.levelData` and stores a bunch of music-specific things, but yes there are other values that are only used in Music Lab that are here too (like - `useSecondaryFinishButton` is technically used in Instructions, but currently only used in the Music Lab HOC freeplay level). Agreed on figuring out a long-term strategy outside of this PR!

Reply:
Thanks to you both for the discussion!